src/test/unit/app/controller/users/EditUserController.ts

```diff
@@ -0,0 +1,291 @@
+
+describe('EditUserController', () => {
+
+  let mockApi: {
+    getUserByEmail: (email: string, accessToken: string) => Promise<User>,
+    updateUserDetails(userId: string, forename: string, surname: string, accessToken: string): Promise<User>,
+    grantUserRole(userId: string, role: object, accessToken: string): Promise<User>,
+    removeUserRole(userId: string, roleName: string, accessToken: string): Promise<User>
+  };
+
+  const user: User = { email: 'name@test.com', lastName: 'lastName', firstName: 'firstName', roles: ['fact-admin']};
+  const updatedUserName: User = { email: 'name@test.com', lastName: 'newLastName', firstName: 'newFirstName', roles: ['fact-admin']};
+  const updatedUserRole: User = { email: 'name@test.com', lastName: 'newLastName', firstName: 'newFirstName', roles: ['fact-super-admin']};
+  const updatedUserRemovedRole: User = { email: 'name@test.com', lastName: 'newLastName', firstName: 'newFirstName', roles: []};
+  const email = 'name@test.com';
+  const userId = 'testId';
+  const updatedFirstName = 'newFirstName';
+  const updatedLastName = 'newLastName';
+  const role = [{'name': 'fact-admin'}];
+  const updatedRole = [{'name': 'fact-super-admin'}];
+
+  const controller = new EditUserController();
+  const res = mockResponse();
+  const req = mockRequest();
+  CSRF.create = jest.fn().mockReturnValue('validCSRFToken');
+  CSRF.verify = jest.fn().mockReturnValue(true);
+
+  const getUserByEmail: () => User = () => user;
+  const updateUserDetails: () => User = () => user;
+  const grantUserRole: () => User = () => user;
+  const removeUserROle: () => User = () => user;
+
+  beforeEach(() => {
+    mockApi = {
+      getUserByEmail: async (): Promise<User> => getUserByEmail(),
+      updateUserDetails: async (): Promise<User> => updateUserDetails(),
+      grantUserRole: async (): Promise<User> => grantUserRole(),
+      removeUserRole: async (): Promise<User> => removeUserROle()
+    };
+
+    CSRF.create = jest.fn().mockReturnValue('validCSRFToken');
+    CSRF.verify = jest.fn().mockReturnValue(true);
+
+    req.scope.cradle.idamApi = mockApi;
+    req.session.user.accessToken = 'accessToken';
+
+    jest.spyOn(mockApi, 'getUserByEmail');
+    jest.spyOn(mockApi, 'updateUserDetails');
+    jest.spyOn(mockApi, 'grantUserRole');
+    jest.spyOn(mockApi, 'removeUserRole');
+  });
+
+  test('Should render the search user page', async () => {
+
+    await controller.renderSearchUser(req, res);
+
+    const pageData: SearchUserPageData = {
+      errors: Array.of(),
+      userEmail: '',
+      updated: false,
+      userRolesRemoved: false
+    };
+
+    expect(res.render).toBeCalledWith('users/tabs/searchUserContent', pageData);
+  });
+
+  test('Should render the edit user page with user', async () => {
+
+    await controller.renderEditUser(req, res);
+
+    const pageData: EditUserPageData = {
+      errors: Array.of(),
```

Comment:
Is Array.of() the same as [] ? 

Interested :) Not sure which is preferred

Reply:
I've updated to [] since that is what's used elsewhere in the file